### DISCUSSION TITLE
Reduction of msgid in py/pot/po results in an increase of strings on weblate

### DISCUSSION THREAD

**Discussion by buhtz** [WeblateOrg/weblate#9430] [Q&A]
I do use [translate.codeberg.org](https://translate.codeberg.org) as Weblate instance. My project had 354 translatable strings. I reduced that number via modifying the Python sources.

Counting the number of msgid's in the message template (via `cat common/po/messages.pot | grep msgid | wc -l) and in the resulting language file (via `cat common/po/de.po | grep "^msgid" | wc -l`) I got 321. That makes sense to me.

But weblate (after "Update/Pull") do [show me 437 strings](https://translate.codeberg.org/projects/backintime/common/de/). I'm confused.

I noticed that the "old strings" are gone in the pot file but not in the po file. In the po file the old msgid is outcommented starting with `#~msgid`. Maybe this is a reason and has something to do with it?

To increase my confusion. I'm not able to find the "old strings" in weblate. So why does the string counter at weblate increased?

**Comment by buhtz** [comment]
I'm not sure but I have an [idea](https://github.com/bit-team/backintime/pull/1452#issuecomment-1596815885).

The setup of my Weblate project might be unusual. The weblate instance is not able to commit/push/PR into the upstream repo because weblate is hosted at Coderberg.org but the upstream repo is hosted at Microsoft GitHub. I do this semi-manually via [nice script](https://github.com/bit-team/backintime/blob/-/update_language_files.py) cloning the weblate repo and just copy the po files into my upstream repo.

Because of that Weblate shows me one or more "Outgoing commits in the Weblate repository" in the maintenance section. This commits never go out because that Weblate instances has no way to push/commit/PR this.

Because of that I assume that when doing "Update" on Weblate the upstream po files are merged with the current ones in the weblate repo. That causes the problem.

But when I "Reset" the Weblate repo it works (nearly #9431) fine.

**Comment by nijel** [comment]
Weblate can update of PO files for you with an add-on: https://docs.weblate.org/en/latest/admin/addons.html#update-po-files-to-match-pot-msgmerge
